api-server/src/server/utils/get-blocks-challenges-mappings.js

```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line import/no-unresolved
+import curriculum from '../../../../shared/config/curriculum.json';
+
+let challengeMap = null;
```

Comment:
Instead of floating global variables, follow the same pattern we use elsewhere of calling the function in the root of the `boot` setup. We do this a lot in `boot/challenge.js`.

Reply:
Thanks a lot for your feedback, @ShaunSHamilton. It really got me thinking about how we handle global states and inspired me to make some changes to our approach with the `challengeMap`.

I did look into how we do things in `boot/challenge.js` and thought about following that pattern closely. But then, I realized we needed something slightly different for this case. I wanted to create a service that could be used across our codebase without having to rebuild the map every time it's needed. So, I went with a solution where we build the `challengeMap` just once and then use it wherever required. This way, it acts more like a service, ready to provide the map on demand, ensuring we only do the heavy lifting of building it a single time.

I've put these changes into practice, hoping it strikes the right balance between avoiding globals and providing a service that’s easily reusable. It’s all updated and ready for you to take a look. You can check out the service encapsulating design in commit [2e12cc5a63095f31d5998f4c878309e64baaa531] and the usage in commit [8381d82add7343f505d2be787c1c039fcb744bb9].

I really tried to make the best choice for our code’s future flexibility and maintainability. If you have any thoughts or further advice on this, I’m all ears. Your input has been incredibly helpful.

Reply:
Thank you. Here is what I suggest:
1) Move most of the logic into `get-curriculum.js`
  - Nothing fundamentally different is being done between the current two files
2) Replace the "service" pattern with a simple:

```js
function getBlockToChallengeIdMap() {
  // ...
  return blockToChallengeIdMap;
}

export const blockToChallengeIdMap = getBlockToChallengeIdMap();
```

3) Import `blockToChallengeIdMap` into `user.js`
4) Replace the `try...catch` with a simple index and `undefined` check
5) Change the `post` request to a `delete` request
6) Instead of using the url params, expect a JSON body which is an array of blocks
  - This helps when URLs are length-restricted
  - This helps when we want to delete more than one block (e.g. a superblock) progress
 
 
One final thing about the reason to use `export const blockToChallengeIdMap = [];` instead of `export function getOrGenerateMap() {}` is how the garbage collector might respond to thinking the resource (map) is no longer needed. Personally, I know very little about this. So, maybe V8 is smart enough to understand the permanency of the memory 🤷‍♂️

Reply:
Hey @ShaunSHamilton,

A huge thanks for your insightful feedback! I've carefully reviewed your suggestions and integrated them into our project.

Following your advice:
- I've consolidated the logic into `get-curriculum.js` and embraced the straightforward approach of directly exporting `blockToChallengeIdMap`.
  
- Our endpoint now accepts DELETE requests and expects a JSON body with an array of blocks.

- In terms of error handling, I've moved away from the previous `try...catch` approach in favor of more direct index checks and validations for undefined values.

- I've also given some thought to the construction of our response messages. Understanding its potential value for frontend development, I've encapsulated this message-building logic into a separate function. This strategy is designed to offer us the flexibility to refine our responses as needed, or scale it back effortlessly, depending on our needs as we move towards production.

Your feedback has been incredibly valuable, @ShaunSHamilton.
If you have any more suggestions or notice anything that might need a re-look, please don't hesitate to let me know. 
I truly appreciate your guidance and am looking forward to any additional insights you might have.

Thanks again!